### ISSUE TITLE
Included role is erroneously skipped, when the including role depends on another role whose unrelated dependency is skipped

### ISSUE THREAD

**Issue by eerorika** [ansible/ansible#23379]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
include_role

##### ANSIBLE VERSION

```
ansible 2.2.2.0
  config file = 
  configured module search path = Default w/o overrides
```

##### OS / ENVIRONMENT

N/A

##### SUMMARY

Included role is erroneously skipped, when the including role depends on another role whose unrelated dependency is skipped.

Since test_3 is included unconditionally, and the skipped dependency is unrelated, test_3 shouldn't be skipped, but it is.

I get expected output if
  - The condition of the test_2 -> test_4 dependency is True.
  - Or the test_2 -> test_4 dependency is removed.
  - Or the test_1 -> test_2 dependency is removed.

##### STEPS TO REPRODUCE

```sh
tail  **/*.yml
```

```yaml
==> roles/test_1/meta/main.yml <==
---
dependencies:
  - test_2

==> roles/test_1/tasks/main.yml <==
---
- include_role:
    name: test_3

==> roles/test_2/meta/main.yml <==
---
dependencies:
  - {
    role: test_4, # conditional dependency
    when: no      # whose condition evaluates to False
  }

==> roles/test_3/tasks/main.yml <==
---
- debug:
    msg: 3

==> roles/test_4/tasks/main.yml <==
---

==> test.yml <==
---
- hosts: all
  roles:
    - test_1
```

```sh
ansible-playbook -vvvv -i "localhost," -c local test.yml
```

##### EXPECTED RESULTS

```
TASK [test_3 : debug] **********************************************************
ok: [localhost] => {
    "msg": 3
}
```

##### ACTUAL RESULTS

```
No config file found; using defaults
Set default localhost to localhost
Loading callback plugin default of type stdout, v2.0 from /usr/lib64/python2.7/site-packages/ansible/plugins/callback/__init__.pyc

PLAYBOOK: test.yml *************************************************************
1 plays in test.yml

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
Using module file /usr/lib64/python2.7/site-packages/ansible/modules/core/system/setup.py
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: usr
<localhost> EXEC /bin/sh -c 'echo ~ && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/usr/.ansible/tmp/ansible-tmp-1491519674.09-105434180556369 `" && echo ansible-tmp-1491519674.09-105434180556369="` echo /home/usr/.ansible/tmp/ansible-tmp-1491519674.09-105434180556369 `" ) && sleep 0'
<localhost> PUT /tmp/tmp_f8MiY TO /home/usr/.ansible/tmp/ansible-tmp-1491519674.09-105434180556369/setup.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/usr/.ansible/tmp/ansible-tmp-1491519674.09-105434180556369/ /home/usr/.ansible/tmp/ansible-tmp-1491519674.09-105434180556369/setup.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python /home/usr/.ansible/tmp/ansible-tmp-1491519674.09-105434180556369/setup.py; rm -rf "/home/usr/.ansible/tmp/ansible-tmp-1491519674.09-105434180556369/" > /dev/null 2>&1 && sleep 0'
ok: [localhost]

TASK [test_3 : debug] **********************************************************
task path: /home/eero/PycharmProjects/ansible_test/roles/test_3/tasks/main.yml:2
skipping: [localhost] => {
    "changed": false, 
    "skip_reason": "Conditional check failed", 
    "skipped": true
}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0 
```

**Comment by eerorika** [comment]
For what it's worth, I wrote a unit test for this: [test_include_role_deps.py](https://github.com/eerorika/ansible/blob/418a07a4e8f5268c564d582856e89d803914cad0/test/units/playbook/role/test_include_role_deps.py)

**Comment by mario-pranjic** [comment]
Hi,

Any feedback on this bug?

**Comment by sivel** [comment]
This issue was resolved in eeaa67657750005b5ff9c287b1883fc08692814d and included in the v2.3.2 release

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project
